go/pserver/service.go

```diff
@@ -51,6 +56,88 @@ type Service struct {
 	optMap map[string]*optimizer
 }
 
+// CheckpointMeta saves the checkpoint information
+type CheckpointMeta struct {
+	UUID      string `json:"uuid"`
+	MD5       string `json:"md5"`
+	Timestamp string `json:"timestamp"`
+}
+
+// ParameterCheckpoint saves parameter checkpoint
+type ParameterCheckpoint struct {
+	ParameterWithConfig
+	Stat []byte
+}
+
+// Checkpoint saves all parameters' checkpoint
+type Checkpoint struct {
```

Comment:
maybe these things should not be the member of `Checkpoint`. They do not persist in permanent storage.  I move them into Server's argument.